test/unit/reductions/exponentiated_gradient/test_exponentiatedgradient_smoke.py

```diff
@@ -208,14 +208,21 @@ def run_smoke_test(self, data, flipped=False):
         ratio = 1.0
         if "ratio" in data.keys():
             ratio = data["ratio"]
-        expgrad = ExponentiatedGradient(self.learner, constraints=data["cons_class"](ratio=ratio),
-                                        eps=data["eps"])
+
+        eps = 0.01
+        if "eps" in data.keys():
+            eps = data["eps"]
+
+        expgrad = ExponentiatedGradient(self.learner, constraints=data["cons_class"](ratio_bound_slack=eps,
```

Comment:
Hmm in that case, should we add **difference_bound** as an argument to the DP constraint defined in the **smoke_test_data** object? As of now, there's only a ratio parameter that's specified (and which is initialized in both difference_bound and ratio_bound). So if we were to have an if/else statement for ratio_bound and difference_bound then we would have to define ratio_bound and difference_bound parameters instead of ratio in the DP constraint right? 

Otherwise I'm not sure how this would be useful because ratio is initialized

Reply:
I think it makes sense but just to make sure I understood correctly, does this pseudocode look right? 
```python
        if "ratio" in data.keys():
            ratio = data["ratio"]
            ratio_bound_slack = data["eps"]
            expgrad = ExponentiatedGradient(learner, constraints(ratio_bound_slack,ratio), eps)
        else:
            difference_bound = data["eps"]
            expgrad = ExponentiatedGradient(learner,
                                            constraints(difference_bound), eps=data["eps"])
 ```
And by doing this I don't need to create this variable "eps".

Reply:
I would make "constraints" -> "Constraints" so that it's clearer that it's a class constructor, not another kind of method.
Other than that it looks fine to me and is exactly what I meant :-)

Reply:
Sorry maybe that wasn't a great pseudocode. 
This is the new version that hopefully incorporates your feedback :)
``` python
        if "ratio" in data.keys():
            expgrad =  ExponentiatedGradient(self.learner, constraints=data["cons_class"](ratio_bound_slack=data["eps"],
                                                                                          ratio_bound=data["ratio"]), eps=data["eps"])
        else:
            expgrad = ExponentiatedGradient(self.learner, constraints=data["cons_class"](difference_bound=data["eps"]),eps=data["eps"])
```